### ISSUE TITLE
DB Sorting in invoices tab produces an error

### ISSUE THREAD

**Issue by blueforce** [akeeba/akeebasubs#263]
Hi I just tested the akeeba subs 5.0.5 with J! 3.5 B5 and PHP 7.0.4 environment.

If I open the invoices tab I get this error:
` 1054 Unknown column 'Array' in 'where clause' SQL=SELECT * FROM #__akeebasubs_subscriptions WHERE 'akeebasubs_subscription_id' = Array AND ('akeebasubs_subscription_id' IN ('123','456','789' ......)) ORDER BY 'akeebasubs_subscription_id' DESC`

Can anyone reproduce this DB error? all other listings seems to work.

**Comment by nikosdion** [comment]
Cannot reproduce. I suspect that you just need to log out and log back in again.

**Comment by blueforce** [comment]
Hi Nik, thank you...
Nope, the problem still remains. But it's ok for the moment. I look forward to find the issue...

**Comment by nikosdion** [comment]
What are the exact steps you're following to reproduce this after clearing logging in?

**Comment by blueforce** [comment]
I go to components > akeebasubs > invoices > invoices

this view: administrator/index.php?option=com_akeebasubs&view=invoices
the error appears directly, all other views working properly (only this view sucks)

**Comment by nikosdion** [comment]
Cannot reproduce with current development branch of Akeeba Subscriptions and FOF 3.

**Comment by chivitli** [comment]
I've seen something similar (or it was the same) in Akeeba subs 5.0.5, also on the invoices tab. I tried to track it and saw that `akeebasubs_subscription_id` was returned as an array when it was supposed to return a string/int, or be empty. However, the problem disappeared by itself, perhaps after I cleared the session cookie, I don't remember now.

**Comment by nikosdion** [comment]
That's exactly why I asked to clear cookies and what are the exact steps being followed. The model state
akeebasubs_subscription_id would be an array if any code was expecting to use the Filters behavior in the Invoices model and used the array format to that effect. However, the
akeebasubs_subscription_id field is explicitly blacklisted from the Filters behavior exactly because we don't want the behavior to handle it, we want our code to do it. Any code which was using an array state was replaced in 5.0.1. Apart from a session stuck at the pre-5.0.1 days I can't see any other reason why you would see this error.

**Comment by chivitli** [comment]
Logout/in didn't help, but clearing cookies yes. I saw it after upgrade from version 4, and since it never appeared again, you're probably right about the cause. Perhaps to remove it from the cookie automatically upon upgrade from a previous version is a good idea?

**Comment by nikosdion** [comment]
The model state is saved in the session data. The session data –as the name implies– is NOT supposed to survive a session reset which is exactly what is supposed to happen when you log out and log back in. Am I right in assuming that you are not using the database session handler but something like APC or Memcache on the affected site?

**Comment by chivitli** [comment]
In my case (**and I may be wrong as it was a few weeks ago**), I thought I found it in the cookie, not in the session, sorry for the initial confusion with my message (I really need to sleep). But I don't trust my memory, so please forget about it - it's possible I didn't even do a logout but just cleared cookies and logged back in. Whatever the case was, you're right about the cause.

**Comment by nikosdion** [comment]
No way it was in the cookie. All Model state is stored in the session. You can check the code yourself ;) Also, I've never written a Joomla! extension which uses persistent cookies (except one particular feature in Admin Tools which we've pretty much removed) because of the problems introduced by said kind of cookies.

**Comment by chivitli** [comment]
My memory, as suspected, is not so good :) I didn't bother reporting the issue at the time as it disappeared and I couldn't reproduce it again, and as you can see in the meantime I forgot how exactly it went ;)
Perhaps you can add a notice for upgrades from old versions?

**Comment by nikosdion** [comment]
Detecting an upgrade from an old version is more difficult than letting people log out and back in :)

**Comment by blueforce** [comment]
So I just installed the newest Version from yesterday Version 5.1.1
But the problem still remains. I have no clue, what can/should I do, to fix this problem...

Do I understand that right? If all subscribers comes back and logged in once again, the failure is solved?

**Comment by nikosdion** [comment]
Nope. Clear YOUR cookies. On YOUR browser. Then YOU log back in. And do try to copy files manually in case Joomla! is forgetting to replace some file.

**Comment by blueforce** [comment]
Ok all clear! I del MY cookies :) – that solved not may problem, but anyway:
Wich files should I copy from to wehere? Sorry for my question but I really have no clue what happend here... :)

**Comment by nikosdion** [comment]
Component files to component folder. What you are experiencing cannot happen because of the way the Invoices model is written the last few months.

**Comment by blueforce** [comment]
So, now I know a little bit more!

1. I rewarded my J! from 3.5rc4 back to 3.4.8
2. The failure is gone... :)
3. I updated to the newest Subs 5.1.1 in 3.4.8, all looks fine
4. I installed the 3.5.RC4 again the subs is broken:
 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'TABE `#__akeebasubs_levels` DROP KEY `slug`' at line 1 SQL=ALTER TABE `#__akeebasubs_levels` DROP KEY `slug`;

Conclusion:
I hope J! will never release the 3.5... it's so bugge for the moment
